### DISCUSSION TITLE
Post deployment hook and environment variable

### DISCUSSION THREAD

**Discussion by rajyraman** [Azure/azure-dev#2603] [Q&A] [answered]
When I provision and deploy a Function App, post deployment hook can access the FUNCTIONS_NAME output environment variable.

![image](https://github.com/Azure/azure-dev/assets/5035266/1a631ff7-c877-44ef-9537-52d13d9124b2)

![image](https://github.com/Azure/azure-dev/assets/5035266/0d6d5ed0-ab5f-4bc8-9287-6dc563b9d97e)

![image](https://github.com/Azure/azure-dev/assets/5035266/09840dcc-966b-4765-96fe-0dfa39a2f7ef)

But if I just do a deployment I cannot seem to access the environment variable.

![image](https://github.com/Azure/azure-dev/assets/5035266/c503ff17-f0af-41eb-b8a7-4ed50fbff236)

Is there anyway to get hold of the Endpoint URL in the postdeployment hook?

Maybe related to #2389

**Comment by rajyraman** [comment]
~Closing as per https://github.com/Azure/azure-dev/discussions/1613#discussioncomment-6661429. There is no need to use azd env get-values as all the environment variables are injected into the context.~

**Reply by rajyraman** [reply]
The environment variables don't seem to be coming through if I just do deploy. So , I am reopening this discussion @vhvb1989

**Reply by vhvb1989** [reply]
Looks like a bug.
@wbreza any ideas?

**Reply by wbreza** [reply, marked as answer]
It appears that this deployment is occurring with a Github action?  If `azd provision` is not called, then the outputs from the Bicep provisioning are not persisted in Github anywhere.  To ensure that all Bicep outputs are available for `azd` try calling `azd env refresh` before `azd deploy`.  This will hydrate all the bicep outputs and make them available as environment variables for your hooks.

**Reply by rajyraman** [reply]
@wbreza - That worked! I had to refresh the env in predeployment hook and then able to use the env variable in the post deployment hook. I added a condition in pre-deployment hook so that the refresh runs only in CI. Thank you Wallace and Victor.

![image](https://github.com/Azure/azure-dev/assets/5035266/fbf4e29e-f761-4626-8412-0a606d6c5b20)
